platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/components/StateAndRound.java

```diff
@@ -26,4 +26,16 @@
  * @param reservedSignedState the state
  * @param round               the round that caused the state to be created
  */
-public record StateAndRound(@NonNull ReservedSignedState reservedSignedState, @NonNull ConsensusRound round) {}
+public record StateAndRound(@NonNull ReservedSignedState reservedSignedState, @NonNull ConsensusRound round) {
+    /**
+     * Make an additional reservation on the reserved signed state
+     *
+     * @param reservationReason the reason for the reservation
+     * @return this
+     */
+    public StateAndRound makeAdditionalReservation(@NonNull final String reservationReason) {
+        reservedSignedState.getAndReserve(reservationReason);
+
+        return this;
+    }
```

Comment:
This is a bit of an odd pattern. The idea behind the state reservation pattern is that each reservation should have an object that tracks the reservation specifically. That way if we ever have a state leak, we can turn on the debug flag and see exactly which reservation we never released. With the pattern used here, we will see some `ReservedSignedState` objects that never get released, and others that get released multiple times. Even if the reference count ends up being the same, the debug trace will show something different.

A better way to do this would be to have this method return a copy of itself but with its own reservation.